go/vt/sqlparser/comments.go

```diff
@@ -252,3 +260,32 @@ func (d CommentDirectives) IsSet(key string) bool {
 	}
 	return false
 }
+
+// SkipQueryPlanCacheDirective returns true if skip query plan cache directive is set to true in query.
+func SkipQueryPlanCacheDirective(stmt Statement) (skipQuerPlanCacheDirective bool) {
+	switch stmt := stmt.(type) {
+	case *Select:
```

Comment:
Could you not DRY this by doing:

```
case *Select, *Insert, *Update, *Delete:
   ...
```

I'm not sure if that works for type switches though.